PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,322 @@
+import time
+from tkinter import *
+
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+
+class Stack(VisualizationApp):
+    nextColor = 0
+    MAX_ARG_WIDTH = 8
+    MAX_SIZE = 8
+    CELL_WIDTH = 20 * MAX_ARG_WIDTH
+    CELL_HEIGHT = 40
+    CELL_BORDER = 2
+    CELL_BORDER_COLOR = 'black'
+    STACK_X0 = 300
+    STACK_Y0 = 400
+
+    def __init__(self, size=MAX_SIZE, maxArgWidth=MAX_ARG_WIDTH, title="Stack", **kwargs):
+        super().__init__(title=title, maxArgWidth=maxArgWidth, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = []
+        self.buttons = self.makeButtons()
+        self.maxArgWidth = maxArgWidth
+        self.display()
+
+    def __str__(self):
+        return str(self.list)
+
+    # Create an index arrow to point at an indexed cell with an optional name label
+    def createIndex(self, index, name=None):
+        cell_coords = self.cellCoords(index)
+        cell_center = self.cellCenter(index)
+        x0 = self.STACK_X0 - self.CELL_WIDTH * 4 // 5
+        x1 = self.STACK_X0 - self.CELL_WIDTH * 3 // 10
+        y0 = y1 = cell_coords[1] + self.CELL_HEIGHT // 2
+
+        return self.drawArrow(
+            x0, y0, x1, y1, self.VARIABLE_COLOR, self.VARIABLE_FONT, name=("item", index))
```

Comment:
This is odd.  No matter what name is passed as argument to `createIndex()` the name passed to `drawArrow()` is `("item", index)`.  I think the caller creating the index needs to be able to control the name.